### PR TITLE
Change mixnode, gateway, network and nym-api binary licenses to GPLv3

### PR DESCRIPTION
**Pull request by mmsinclair**
Changes the licenses for the binaries that run Nym nodes and APIs to GPLv3

### CODE REVIEW

File: gateway/gateway-requests/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Copyright 2020 - Nym Technologies SA <contact@nymtech.net>
-# SPDX-License-Identifier: Apache-2.0
+# SPDX-License-Identifier: GPL-3.0-only
```
Comment (jstuczyn) [inline gateway/gateway-requests/Cargo.toml]: you don't want to be changing this one, it's part of client library to interact with the gateway (and not the actual gateway). this comment applies to everything in `gateway-requests`

File: nym-api/nym-api-requests/src/coconut.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2022 - Nym Technologies SA <contact@nymtech.net>
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (jstuczyn) [inline nym-api/nym-api-requests/src/coconut.rs]: same problem here as with gateway-requests

**Review by jstuczyn — CHANGES_REQUESTED** [review]

**Comment by octol** [comment]
IMHO we should add the license to the Cargo.toml files as well

File: gateway/gateway-requests/Cargo.toml (hunk 1 above)
Comment (mmsinclair) [reply gateway/gateway-requests/Cargo.toml]: Done

File: nym-api/nym-api-requests/src/coconut.rs (hunk 2 above)
Comment (mmsinclair) [reply nym-api/nym-api-requests/src/coconut.rs]: Done

**Comment by mmsinclair** [comment]
> IMHO we should add the license to the Cargo.toml files as well

Good plan, added this

**Review by jstuczyn — APPROVED** [review]

**Comment by numbleroot** [comment]
Hi! Would it make sense to add the GPLv3 license file to the other licenses distributed as part of the Nym repository in the LICENSES folder?

**Comment by timkuijsten** [comment]
Can I read somewhere about the reasons for the license change?

**Comment by mmsinclair** [comment]
> Hi! Would it make sense to add the GPLv3 license file to the other licenses distributed as part of the Nym repository in the LICENSES folder?

thanks @numbleroot that was an oversight, will add it

**Comment by mmsinclair** [comment]
> Can I read somewhere about the reasons for the license change?

Good question @timkuijsten - we are adding functionality to the gateways to accept wireguard connections, so that users can choose a less-private but faster mode for certain types of traffic, and we will be re-using some libraries that are GPLv3 licensed to manage tunnel devices, hence the change for the binaries.

Please note, the core of Nym's libraries will remain Apache 2.0, so that they can be easily used in software.

Also, bear in mind that we are part-way through smooshing all of Nym's binaries together into a single binary that offers all functionality that will simply be called `nym-node`. The `nym-node` binary will be GPLv3 licensed.

The standalone Nym clients (SOCKS5/websockets), will still remain Apache 2.0 licensed, so that they can be easily included in software.

Hope that clears things up 😁

**Pull request merged**